src/main/resources/api/service_openapi.yaml

```diff
@@ -1613,6 +1645,17 @@ components:
         cloningInstructions:
           $ref: '#/components/schemas/CloningInstructionsEnum'
 
+    UpdateControlledResourceRequestBody:
+      type: object
+      description: Common request body for updating a controlled resource
+      properties:
+        name:
+          description: Optional. New name to give to this resource. The resource name will not be updated if this is omitted.
```

Comment:
Does `null` have different semantics from the key being omitted?

Reply:
I think the problem is that when reading a request object we can't tell the difference between a key being present and `null` vs. being absent, so semantically they're the same here.

Reply:
Oh, does Swagger insert all the missing keys and give them a null value? That would do it. I was thinking it was like a map where you could use `contiansKey` to distinguish the two cases.

Reply:
It's more like we get an `UpdateControlledResourceRequestBody` object passed to our Controller method, and when `UpdateControlledResourceRequestBody.getName()` returns null, we can't tell whether that means the value was omitted, or included and set to null

Reply:
Yeah, it sounds like we need a free-form map if we want the patch semantics.